packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-has-one-relationship/BookUpdateForm.jsx

```diff
@@ -0,0 +1,391 @@
+    setName(initialValues.name);
+    setPrimaryAuthor(primaryAuthor);
+    setCurrentPrimaryAuthorValue(undefined);
+    setCurrentPrimaryAuthorDisplayValue(undefined);
+  };
+
+  // fetch model data to fill form with preset values
+  const [bookRecord, setBookRecord] = React.useState(book);
+  React.useEffect(() => {
+    const queryData = async () => {
+      const record = id ? await DataStore.query(Book, id) : book;
+      setBookRecord(record);
+      // get existing relationship record
+      const existingAuthor = record ? await DataStore.query(Author, (a) => a.id.eq(record.authorId)) : primaryAuthor;
+      setPrimaryAuthor(existingAuthor);
+    };
+    queryData();
+  }, [id, book]);
+
+  React.useEffect(resetStateValues, [bookRecord]);
+
+  // displayValue (string) and value (Model) must be set separately
+  const [currentPrimaryAuthorValue, setCurrentPrimaryAuthorValue] = React.useState(undefined);
+
+  const [currentPrimaryAuthorDisplayValue, setCurrentPrimaryAuthorDisplayValue] = React.useState(undefined);
+  const primaryAuthorRef = React.createRef();
+
+  const validations = {
+    name: [],
+    primaryAuthor: [],
+  };
+
+  // new arg so we're validating string, not Model
+  const runValidationTasks = async (fieldName, currentValue, getDisplayValue) => {
+    const value = getDisplayValue ? getDisplayValue(currentValue) : currentValue;
+    let validationResponse = validateField(value, validations[fieldName]);
+    const customValidator = fetchByPath(onValidate, fieldName);
+    if (customValidator) {
+      validationResponse = await customValidator(value, validationResponse);
+    }
+    setErrors((errors) => ({ ...errors, [fieldName]: validationResponse }));
+    return validationResponse;
+  };
+  return (
+    <Grid
+      as="form"
+      rowGap="15px"
+      columnGap="15px"
+      padding="20px"
+      onSubmit={async (event) => {
+        event.preventDefault();
+        let modelFields = {
+          name,
+          primaryAuthor,
+        };
+        const validationResponses = await Promise.all(
+          Object.keys(validations).reduce((promises, fieldName) => {
+            if (Array.isArray(modelFields[fieldName])) {
+              promises.push(
+                ...modelFields[fieldName].map((item) =>
+                  // pass in getDisplayValue callback to validate string, not Model
+                  runValidationTasks(fieldName, item, getDisplayValue[fieldName]),
+                ),
+              );
+              return promises;
+            }
+            promises.push(runValidationTasks(fieldName, modelFields[fieldName], getDisplayValue[fieldName]));
+            return promises;
+          }, []),
+        );
+        if (validationResponses.some((r) => r.hasError)) {
+          return;
+        }
+        if (onSubmit) {
+          modelFields = onSubmit(modelFields);
+        }
+        try {
+          const original = await DataStore.query(Book, id);
+          await DataStore.save(
```

Comment:
This just saves without needing to do anything else with `primaryAuthor`?

Reply:
I've updated the code slightly to `primaryAuthor: primaryAuthor?.id,` so that it captures the `id` in `modelFields` and updates correctly